test/recipes/default/user_test.rb

```diff
@@ -0,0 +1,71 @@
+describe group('stanley') do
+  it { should exist }
+end
+
+describe user('stanley') do
+  it { should exist }
+  its('group') { should eq 'stanley' }
+  its('home') { should eq '/home/stanley' }
+end
+
+describe passwd.users(/stanley/) do
+  its('homes') { should eq ['/home/stanley'] }
+  its('shells') { should eq ['/bin/bash'] }
+end
+
+describe directory('/home/stanley/.ssh') do
+  it { should exist }
+  its('owner') { should eq 'stanley' }
+  its('group') { should eq 'stanley' }
+  it { should be_readable.by('owner') }
+  it { should_not be_readable.by('group') }
+  it { should be_writable.by('owner') }
+  it { should_not be_writable.by('group') }
+  it { should be_executable.by('owner') }
+  it { should_not be_executable.by('group') }
+end
+
+describe file('/home/stanley/.ssh/id_rsa') do
+  it { should exist }
+  its('owner') { should eq 'stanley' }
+  its('group') { should eq 'stanley' }
+  it { should be_readable.by('owner') }
+  it { should be_readable.by('group') }
+  it { should be_writable.by('owner') }
+  it { should_not be_writable.by('group') }
+  it { should_not be_executable.by('owner') }
+  it { should_not be_executable.by('group') }
+end
+
+describe file('/home/stanley/.ssh/id_rsa.pub') do
+  it { should exist }
+  its('owner') { should eq 'stanley' }
+  its('group') { should eq 'stanley' }
+  it { should be_readable.by('owner') }
+  it { should be_readable.by('group') }
+  it { should be_readable }
+  it { should be_writable.by('owner') }
+  it { should_not be_writable.by('group') }
+  it { should_not be_executable.by('owner') }
+  it { should_not be_executable.by('group') }
+end
+
+describe file('/home/stanley/.ssh/authorized_keys') do
+  it { should exist }
+  its('owner') { should eq 'stanley' }
+  its('group') { should eq 'stanley' }
+  it { should be_readable.by('owner') }
+  it { should be_readable.by('group') }
+  it { should be_readable }
+  it { should be_writable.by('owner') }
+  it { should_not be_writable.by('group') }
+  it { should_not be_executable.by('owner') }
+  it { should_not be_executable.by('group') }
+end
+
+describe file('/etc/sudoers.d/stanley') do
+  it { should exist }
+  its('content') { should match(%r{#\s.*?This\s.*?file\s.*?is\s.*?managed\s.*?by\s.*?Chef.}) }
```

Comment:
`\s.*?`?
Not sure what the purpose of this is for the regex
You should be able to just use a space or `\s`